pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/SingleConnectionBrokerRequestHandler.java

```diff
@@ -106,13 +106,36 @@ protected BrokerResponseNative processBrokerRequest(long requestId, BrokerReques
       serverBrokerRequest.getPinotQuery().putToQueryOptions(CommonConstants.Broker.Request.TRACE, "true");
     }
 
+    // broker short-circuit for `select` and `select distinct` queries without `ORDER BY`
+    boolean orderByPresent = false;
```

Comment:
Checking if select is present is not enough to determine if the query is select/select distinct. We can use `!QueryContextUtils.isAggregationQuery(query) && query.getOrderByExpressions() == null`, but we will need to first construct the `QueryContext` from the `PinotQuery`. This conversion is not very cheap, so we should try to do this conversion only once in the request handler, and pass it to the reducer

Reply:
Thanks Jackie, it was very helpful.